docs/technical_reference/unit_models/nanofiltration_0D.rst

```diff
@@ -0,0 +1,72 @@
+Nanofiltration (0D)
+====================
+This nanofiltration (NF) unit model is suitable for non-predictive case studies where the user wishes to specify constant recovery fractions
+whilst preserving electroneutrality in the permeate stream (optional). A single recovery fraction is assumed for all solvents, whilst
+individual recovery fractions can be set for neutral and monovalent solutes, with the exception of one ion specified for maintaining
+electroneutrality. By default, complete rejection of all multi-valent solutes is assumed (however a single multi-valent ion recovery fraction
+is provided).
+
+Retentate pressure is assumed to be related to feed pressure with an optional pressure drop whilst permeate pressure is assumed to be
+a degree of freedom, and temperature equality is assumed. Temperature and pressure constraints can be removed with configuration arguments.
+
+This model assumes supports a single liquid phase only and assumes steady-state.
+
+.. index::
+   pair: watertap.unit_models.nanofiltration_0D;nanofiltration_0D
+
+.. currentmodule:: watertap.unit_models.nanofiltration_0D
+
+Degrees of Freedom
+------------------
+The ``Nanofiltration0D`` model has the following degrees of freedom
+
+   * inlet state
+   * permeate pressure
+   * solvent recovery fraction (``solvent_recovery``)
+   * solute recovery fractions (``solute recovery``) for all solutes EXCEPT the one identified as as the electroneutrality species.
+
+The following additional degrees of freedom  may exist depending on configuration options
+
+   * retentate pressure drop (``deltaP``)
+   * multivalent ion recovery (``multivalent_recovery``). Always present but fixed to 1e-10 by default.
+
+Model Structure
+---------------
+The Nanofiltration0D model consists of three state blocks (``properties_in``, ``properties_retentate`` and ``properties_permeate``).
+
+.. _NF0D_variables:
+
+Variables
+---------
+
+.. csv-table::
+   :header: "Description", "Symbol", "Variable Name", "Index", "Units"
+
+   "Solvent recovery", ":math:`R_{solvent}`", "solvent_recovery", None, ":math:`\text{dimensionless}`"
+   "Solute recovery", ":math:`R_j`", "solute_recovery", [j], ":math:`\text{dimensionless}`"
+   "Multivalent ion recovery", ":math:`R_{multi}`", "multivalent_recovery", None, ":math:`\text{dimensionless}`"
```

Comment:
I decided to only include one of rejection or recovery to avoid any chance of over-specifying the problem - it would be easy to switch to rejection if that would be a better choice. I would generally discourage having both, as someone will attempt to fix both of them and wonder why the model fails to solve.

Reply:
In that case, we should move to rejection.

Reply:
Rejection as in `1- permeate_concentration/feed_concentration`. This should already be in the ZO model.